sonar-batch/src/main/java/org/sonar/batch/cache/ProjectCacheSynchronizer.java

```diff
@@ -63,22 +63,48 @@ public ProjectCacheSynchronizer(QualityProfileLoader qualityProfileLoader, Proje
     this.cacheStatus = cacheStatus;
   }
 
+  private static boolean isToday(Date d) {
+    Calendar c1 = Calendar.getInstance();
+    Calendar c2 = Calendar.getInstance();
+    c2.setTime(d);
+
+    return c1.get(Calendar.DAY_OF_YEAR) == c2.get(Calendar.DAY_OF_YEAR) &&
+      c1.get(Calendar.YEAR) == c2.get(Calendar.YEAR);
+  }
+
+  private static boolean shouldUpdate(Date lastUpdate) {
+    return !isToday(lastUpdate);
+  }
+
   public void load(String projectKey, boolean force) {
     Date lastSync = cacheStatus.getSyncStatus();
+    boolean failOnError = true;
 
     if (lastSync != null) {
-      if (!force) {
+      if (force) {
+        LOG.info("-- Found project [{}] cache [{}], synchronizing data (forced)..", projectKey, lastSync);
+      } else if (shouldUpdate(lastSync)) {
+        LOG.info("-- Found project [{}] cache [{}], synchronizing data..", projectKey, lastSync);
+        failOnError = false;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "failOnError". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1854)